pkg/metrics/gauge_with_timestamp.go

```diff
@@ -0,0 +1,35 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+package metrics
+
+// GaugeWithTimestamp tracks the value of a metric with a given timestamp
+type GaugeWithTimestamp struct {
```

Comment:
This type does exactly the same thing as CountWithTimestamp. Would it make sense to combine them in the same type?

Reply:
Indeed, the only difference is the API type used during flush
https://github.com/DataDog/datadog-agent/blob/ae6fd500cf8e9b1c92c3a4722c7ec37785959225/pkg/metrics/gauge_with_timestamp.go#L28
https://github.com/DataDog/datadog-agent/blob/ae6fd500cf8e9b1c92c3a4722c7ec37785959225/pkg/metrics/count_with_timestamp.go#L28

What we could do is combine them under a common type (named `MetricWithTimestamp`?) that would take the api type on initialisation like this:

```go
// context_metrics.go
func (m ContextMetrics) AddSample(...) error {
                 ...
		case GaugeWithTimestampType:
			m[contextKey] = NewMetricWithTimestamp(APIGaugeType)
		case CountWithTimestampType:
			m[contextKey] = NewMetricWithTimestamp(APICountType)
                 ...
```
What do you think?

Reply:
Done in e5ab818